src/data/feature_index.js

```diff
@@ -194,16 +198,34 @@ class FeatureIndex {
             }
 
             const styleLayer = styleLayers[layerID];
+
             if (!styleLayer) continue;
 
-            const intersectionZ = !intersectionTest || intersectionTest(feature, styleLayer, id);
+            let featureState = {};
+            if (id !== undefined && sourceFeatureState) {
+                // `feature-state` expression evaluation requires feature state to be available
+                featureState = sourceFeatureState.getState(styleLayer.sourceLayer || '_geojsonTileLayer', id);
+            }
+
+            const serializedLayer = serializedLayers[layerID];
+
+            const evaluationParameters = {
+                feature,
+                featureState,
+                availableImages
+            };
```

Comment:
Nit: probably no need to wrap those in an object if you immediately unwrap them in the lines below — separate arguments are fine.